safekeeper/src/pull_timeline.rs

```diff
@@ -503,6 +503,33 @@ pub async fn handle_request(
 
     let check_tombstone = !request.ignore_tombstone.unwrap_or_default();
 
+    if status.timeline_start_lsn == Lsn(0) || status.timeline_start_lsn == status.commit_lsn {
```

Comment:
Why do we look at `commit_lsn`?
Shouldn't we look at flush_lsn?

Reply:
I made it use the same code now that the code that emits the `snapshot is called on uninitialized timeline` error uses.